src/provide-context.test.js

```diff
@@ -0,0 +1,187 @@
+import * as React from "react";
+import provideContext from "./provide-context";
+import createResource from "./create-resource";
+import createSimpleResource from "./create-simple-resource";
+import { assertResourceShape } from "./__mocks__/assert-resource-shape";
+import { render, waitForElement } from "react-testing-library";
+
+describe("provideContext", () => {
+  it("is a curried function with an arity of two", () => {
+    expect(typeof provideContext).toBe("function");
+    expect(provideContext.length).toBe(2);
+
+    const result = provideContext(x => x);
+    expect(typeof result).toBe("function");
+    expect(result.length).toBe(1);
+  });
+
+  it("creates a resource", () => {
+    assertResourceShape(
+      provideContext(
+        x => x,
+        createSimpleResource(() => Promise.resolve("resolved"))
+      )
+    );
+  });
+
+  it("provides the return value of the provider to the selectState function", () => {
+    const value = "abc123";
+    const selectState = jest
+      .fn()
+      .mockImplementation((state, [authToken]) => state);
+    const resource = provideContext(
+      () => value,
+      createResource({
+        selectState,
+        setState: (_, __, value) => value,
+        loader: authToken => Promise.resolve(authToken)
+      })
+    );
+
+    const Example = () => {
+      const [state] = resource.useState();
+      return state;
+    };
+
+    render(
+      <React.Suspense fallback={null}>
+        <Example />
+      </React.Suspense>
+    );
+
+    expect(selectState).toHaveBeenCalledWith(undefined, [value]);
+  });
+
+  it("provides the return value of the provider to the setState function", async () => {
+    const value = "abc123";
+    const setState = jest.fn().mockImplementation((_, __, value) => value);
+    const resource = provideContext(
+      () => value,
+      createResource({
+        selectState: state => state,
+        setState,
+        loader: authToken => Promise.resolve(authToken)
+      })
+    );
+
+    const Example = () => {
+      const [state] = resource.useState();
+      return state;
+    };
+
+    const { getByText } = render(
+      <React.Suspense fallback={null}>
+        <Example />
+      </React.Suspense>
+    );
+
+    await waitForElement(() => getByText(value));
+
+    expect(setState).toHaveBeenCalledWith(undefined, [value], value);
+  });
```

Comment:
**Question**

Are the values only added to the context of `setState` inside of `useState`? For example if someone used `resource.setState` would those have the provided extra context need to passed in?

Reply:
Yes, they are only provided the context in `useState`, but I don't know of any other method besides `refresh` where it would be useful. And even in the case of `refresh` I'm not sure how we could use the function if it's a hook, since hooks can only be used inside a React component. But . I could see how being able to provide that context to `refresh` could be useful though. I'm just not sure how to accomplish it.

Let me know if you have any ideas.